### PR TITLE
Docs

### PR DESCRIPTION
**Pull request by JakeGinnivan**
// @gep13 @GeertvanHorrik 

Adds some new docs in the repo (I plan on turning off the wiki) and also made decent headway into `git init` which makes it easy to edit the configuration. It also has some basic questions to help people decide on the workflow they will use etc.

Feedback would be great

### CODE REVIEW

**Comment by robdmoore** [comment]
Epic! (Looks good)

File: GitVersionCore/Configuration/Wizard/PickBranchingStrategy1Step.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+            }
+
+            return StepResult.InvalidResponseSelected();
+        }
+
+        protected override string GetPrompt(Config config)
+        {
+            return @"GitVersion can try to recommend you a branching strategy based on a few questions. 
+
+Do you need to maintain mutliple versions of your application simultanously in production? (y/n)";
```
Comment (gep13) [inline GitVersionCore/Configuration/Wizard/PickBranchingStrategy1Step.cs]: Slight spelling issue here: http://dictionary.reference.com/browse/simultaneously

**Comment by gep13** [comment]
@JakeGinnivan I really like the Wizard Configuration steps, I think this will be really helpful going forward.

Regarding the docs, isn't this something that we should be putting into the readthedocs pages for this project?  I thought that is where we decided to put this?

**Comment by JakeGinnivan** [comment]
Forgot about the read the docs. Where is the link for it, we don't have it listed anywhere?

Sent from my Windows Phone
________________________________

**Comment by gep13** [comment]
http://gitversion.readme.io/ I think @GeertvanHorrik added you to the list of maintainers/authors, so you should be able to add the docs in there.  All the projects are available:

* gitversion.readme.io
* gitreleasenotes.readme.io
* gitreleasemanager.readme.io

And I think this one: http://gittools.readme.io/ was planned to be the one to join them all together.

**Comment by JakeGinnivan** [comment]
Ah yeah. 

One issue is I did the docs while on a plane. Does readme.io have offline editing support?

**Comment by gep13** [comment]
Good question, in all honesty, I am not sure.  There is some element of syncing with Github for the docs, but I think that is more for documenting API's methods, etc, from code, which is not really what we are after.

File: Docs/configurationOptions.md [hunk 2]
```diff
@@ -0,0 +1 @@
+# Confiuration options
```
Comment (GeertvanHorrik) [inline Docs/configurationOptions.md]: Typo

File: Docs/index.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
+# GitVersion Docs
+GitVersion is a tool to help you achieve Semantic Versioning on your project.
+
+This influences many of the decisions GitVersion has made, please read and understand this page as it will help you start using GitVersion effectively!
+
+## Assumptions/Rules
+### You tag releases when you release, not before
+GitVersion assumes that tags are used to *tag a release* and not used to build a release.
+
+This means that the version is calculated pre-emptively, if you currently tag, build then release that tag then GitVersion will probably not work for you.
```
Comment (GeertvanHorrik) [inline Docs/index.md]: Not sure what this means. We create a release branch, then merge to master, tag and then release. That's working fine here.

Comment (gep13) [inline Docs/index.md]: Actually, yeah, this is exactly what we do on ChocolateyGUI as well with GtiVersion, and this works fine for me currently, on beta 2.

File: Docs/index.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+Read more about the different [configuration options](./configurationOptions.md)
+
+## Output Variables
+We recognise that a single formatted version number does not work for all cases. A simple example is NuGet, it doesn't support SemVer 2.0 meaning that the SemVer of `1.3.5-beta.10+500` needs to be formatted as `1.3.5-beta0010` so it will sort properly.
+
+You can just run `GitVersion.exe` in your repository to see what variables are available (by default a json object is returned).
+
+## Exe or MSBuild Task
+There are two ways to consume GitVersion, the first is by running GitVersion.exe. The second is an MSBUild task. The MSBuild task is really easy to get up and running, simply install GitVersionTask from NuGet and it will integrate into your project and write out variables to your build server if it's running on one. The exe offers more options and works for not just .net projects.
```
Comment (GeertvanHorrik) [inline Docs/index.md]: MSBUild => MSBuild (U shouldn't be capitalized)

File: Docs/index.md (hunk 3 above)
Comment (JakeGinnivan) [inline Docs/index.md]: You tag release/hotfix/support branches before releasing too?

Could expand this to explain GitFlow, but completing a gitflow release is a specific case, not the rule

Comment (JakeGinnivan) [inline Docs/index.md]: And the rule below explains why your workflow works fine too

Comment (gep13) [inline Docs/index.md]: No, I personally only ever tag the master branch.

File: Docs/versionSources.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+ - Calculate the build metadata (everything after the +) and append to the calcuated version
+
+## Version Sources
+### Highest Accessible Tag
+GitVersion will find all tags on the current branch and return the highest one.
+
+Will increment: true
+
+### Version in branch name
+If the branch has a version in it, then that version will be returned
```
Comment (GeertvanHorrik) [inline Docs/versionSources.md]: Missing .

File: Docs/versionSources.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+Will increment: false
+
+### Merge message
+If a branch with a version number in it is merged into the current branch, that version number will be used.
+
+Will increment: false
+
+### GitVersionConfig.yaml
+If the next-version property is specified in the config file, it will be used as a version source.
```
Comment (GeertvanHorrik) [inline Docs/versionSources.md]: Put * around next-version

File: GitVersionCore/Configuration/Wizard/EditConfigStep.cs [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+                    return StepResult.Ok();
+            }
+            return StepResult.Ok();
+        }
+
+        protected override string GetPrompt(Config config)
+        {
+            return string.Format(@"Which would you like to change?
+
+0) Save changes
```
Comment (GeertvanHorrik) [inline GitVersionCore/Configuration/Wizard/EditConfigStep.cs]: Save changes and exit

File: GitVersionCore/Configuration/Wizard/PickBranchingStrategy1Step.cs (hunk 1 above)
Comment (GeertvanHorrik) [inline GitVersionCore/Configuration/Wizard/PickBranchingStrategy1Step.cs]: And mutliple should be fixed as well ;-)

File: GitVersionCore/Configuration/Wizard/PickBranchingStrategy2Step.cs [hunk 8]
```diff
@@ -0,0 +1,39 @@
+namespace GitVersion
+{
+    using System;
+    using System.Collections.Generic;
+
+    public class PickBranchingStrategy2Step : ConfigInitWizardStep
```
Comment (GeertvanHorrik) [inline GitVersionCore/Configuration/Wizard/PickBranchingStrategy2Step.cs]: Maybe rename "BranchingStrategy2" to PickBranchingStrategryGitHubFlow, same for v1

File: GitVersionCore/Configuration/Wizard/PickBranchingStrategy2Step.cs [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+    {
+        protected override StepResult HandleResult(string result, Queue<ConfigInitWizardStep> steps, Config config)
+        {
+            switch (result.ToLower())
+            {
+                case "y":
+                    Console.WriteLine("GitFlow is likely a good fit, the 'develop' branch can be used " +
+                                      "for active development while stabilising the next release.");
+                    Console.WriteLine();
+                    Console.WriteLine("GitHubFlow is designed for a lightwieght workflow where master is always " +
```
Comment (GeertvanHorrik) [inline GitVersionCore/Configuration/Wizard/PickBranchingStrategy2Step.cs]: lightweight

File: GitVersionCore/Configuration/Wizard/PickBranchingStrategy2Step.cs [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+                                      "good to deploy to production and feature branches are used to stabilise " +
+                                      "features, once stable they can be released in the next release");
```
Comment (GeertvanHorrik) [inline GitVersionCore/Configuration/Wizard/PickBranchingStrategy2Step.cs]: "once stable they are merged to master and made available in the next release"

File: GitVersionCore/Configuration/Wizard/PickBranchingStrategy2Step.cs (hunk 8 above)
Comment (JakeGinnivan) [inline GitVersionCore/Configuration/Wizard/PickBranchingStrategy2Step.cs]: The class names I have picked are terrible.. I agree. Will have another go at that tonight

File: GitVersionCore/Configuration/Wizard/PickBranchingStrategy3Step.cs [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+    {
+        protected override StepResult HandleResult(string result, Queue<ConfigInitWizardStep> steps, Config config)
+        {
+            switch (result.ToLower())
+            {
+                case "y":
+                    Console.WriteLine("GitFlow could be a better fit than GitHubFlow for you.");
+                    Console.WriteLine();
+                    Console.WriteLine("GitVersion increments the SemVer for each commit on the develop branch by default, " +
+                                      "this means all packages build from develop can be published to a single NuGet feed.");
```
Comment (GeertvanHorrik) [inline GitVersionCore/Configuration/Wizard/PickBranchingStrategy3Step.cs]: build == built?

File: GitVersionCore/Configuration/Wizard/SetBranchIncrementMode.cs [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+                    steps.Enqueue(new ConfigureBranch(name, branchConfig));
+                    return StepResult.Ok();
+            }
+
+            return StepResult.InvalidResponseSelected();
+        }
+
+        protected override string GetPrompt(Config config)
+        {
+            return string.Format(@"What do you want the increment mode for {0} to be
```
Comment (GeertvanHorrik) [inline GitVersionCore/Configuration/Wizard/SetBranchIncrementMode.cs]: Add question mark to the end of the sentence

File: GitVersionCore/Configuration/Wizard/SetBranchTag.cs [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+    using System.Collections.Generic;
+
+    public class SetBranchTag : ConfigInitWizardStep
+    {
+        string name;
+        readonly BranchConfig branchConfig;
+
+        public SetBranchTag(string name, BranchConfig branchConfig)
+        {
+            this.branchConfig = branchConfig;
```
Comment (GeertvanHorrik) [inline GitVersionCore/Configuration/Wizard/SetBranchTag.cs]: You use name / branchconfig everywhere (right order), make it the same please :-)

File: GitVersionCore/Configuration/Wizard/SimpleOrTutorialStep.cs [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+            {
+                steps.Enqueue(new EditConfigStep());
+            }
+
+            return StepResult.Ok();
+        }
+
+        protected override string GetPrompt(Config config)
+        {
+            return "Would you like to run an extended init (more like a tutorial) (y/n)";
```
Comment (GeertvanHorrik) [inline GitVersionCore/Configuration/Wizard/SimpleOrTutorialStep.cs]: Add question mark

**Comment by GeertvanHorrik** [comment]
@JakeGinnivan , my compliments, the init stuff looks really nice and easy to understand for new users.

**Comment by JakeGinnivan** [comment]
@GeertvanHorrik thanks for the review. Will take another pass at it tonight and see if I can fix a few of the issues you spotted

File: Docs/index.md (hunk 3 above)
Comment (GeertvanHorrik) [inline Docs/index.md]: We tag release branches as well (to create beta versions of upcoming releases).

Comment (JakeGinnivan) [inline Docs/index.md]: I will rethink this section..

File: GitVersionCore/Configuration/Wizard/PickBranchingStrategy2Step.cs (hunk 10 above)
Comment (JakeGinnivan) [inline GitVersionCore/Configuration/Wizard/PickBranchingStrategy2Step.cs]: :+1:

**Pull request merged**